### PR TITLE
Upgrade to Apollo Client 3.0 (@apollo/client)

### PR DESCRIPTION
**Pull request by hwillson**
This PR updates the fullstack tutorial to use `@apollo/client`, which means the following deps have been removed:

- `apollo-client`
- `apollo-cache-inmemory`
- `apollo-link`
- `apollo-link-http`
- `graphql-tag`
- `@apollo/react-hooks`
- `@apollo/react-testing`

This PR is ready for review, but we shouldn't merge it into `master` until the fullstack tutorial web content has been updated to match, and AC 3 is final (so we can drop the `@apollo/client@beta` reference).

One quick thing to note - it looks like the current `master` version of this project has a few issues, that are causing broken tests. I've addressed these issues in this PR:

- This PR removes `final/client/src/pages/__tests__/__generated__/IsUserLoggedIn.ts`, which appears to have been added by codegen. This file isn't needed, and it's being picked up by Jest when trying to run tests, and leading to a test error (since it isn't a test file). I've removed the file for now, but we might want to also either update the `apollo:codegen` command to exclude this file from codegen, or add a custom create-react-app Jest override that excludes this file from testing.
- It looks like launch ID's are now `string`'s in the Typescript types instead of numbers, which means some of the existing tests were failing due to a mismatch of trying to use both numbers and strings for launch ID's in mocked responses. I'm not sure if the `string` launch ID type change was intentional, but I've adjusted the launch ID's in failing tests to be strings to make the tests pass.

(cc @benjamn @jbaxleyiii)

### CODE REVIEW

**Comment by stemmlerjs** [comment]
Excellent stuff, @hwillson! Thanks for all the hard work on this. Unrelated to your changes, I just tested out the starting project on the client-side and added the `LaunchTile` fragment to `launches.tsx`. It was missing from the `cart-items.tsx` file and throwing errors for users just starting up the tutorial.

I also tested out the entire story from front to back in the `final/` folder, and it looks good to me.

**Comment by hwillson** [comment]
Thanks very much for the review and the start fix @stemmlerjs!

**Comment by lifeiscontent** [comment]
@hwillson please fix this issue before you release 3.0

https://github.com/apollographql/react-apollo/pull/3815

File: final/client/src/pages/cart.tsx [hunk 1]
```diff
@@ -19,14 +18,14 @@ const Cart: React.FC<CartProps> = () => {
...
-  
+
   if (loading) return <Loading />;
   if (error) return <p>ERROR: {error.message}</p>;
 
   return (
     <Fragment>
       <Header>My Cart</Header>
-      {!data || !!data && data.cartItems.length === 0 ? (
+      {!data || (!!data && data.cartItems.length === 0) ? (
```
Comment (lifeiscontent) [inline final/client/src/pages/cart.tsx:28]:
```suggestion
      {!data || (data && data.cartItems.length === 0) ? (
```

File: final/client/src/pages/launches.tsx [hunk 2]
```diff
@@ -59,33 +59,23 @@ const Launches: React.FC<LaunchesProps> = () => {
...
-          >
-            Load More
-          </Button>
-        )}
+      {data.launches && data.launches.hasMore && (
+        isLoadingMore
+          ? <Loading />
+          : <Button
+              onClick={async () => {
+                setIsLoadingMore(true);
```
Comment (lifeiscontent) [inline final/client/src/pages/launches.tsx:67]: @hwillson why are you choosing to use `useState` vs fetchMore here?

File: final/client/src/pages/cart.tsx (hunk 1 above)
Comment (hwillson) [reply final/client/src/pages/cart.tsx]: Thanks @lifeiscontent - I'll tweak this a bit.

File: final/client/src/pages/launches.tsx (hunk 2 above)
Comment (hwillson) [reply final/client/src/pages/launches.tsx:67]: @lifeiscontent can you explain your question a bit more? I'm using state to control a more fine-grained loading indicator, when people click to load more entries using `fetchMore`.

**Comment by hwillson** [comment]
This PR is now up to date with the AC3 RC API. We'll need to wrangle the [FS tutorial content](https://www.apollographql.com/docs/tutorial/introduction/) into shape next, to match the changes in this PR.

**Pull request merged**